### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by IlmariKu**
This pull-request is about cleaning up duplicate functionalities to reduce clutter. The project-root folder is full of stuff and there's a lot of copy-pasting around, making everything hard to follow.

- Helpers to have their own folder
- Start scripts now utilize some shared behavior (options-object, 
- Instructions.txt's to be loaded from separate folder.
- Unify cmd's under one command (so you now need to pass a provider as first argument), couldn't test, no windows
- Whitespace fixes (set up in editor)
- Scripts moved to scripts-folder and instructions added if one wants to use them for their specific system.

Things I couldn't test:
- Other models (besides Azure). GUI works.
- CMD's

### CODE REVIEW

**Comment by IlmariKu** [comment]
Maybe if we would add all the different .cmd, .bat and .sh scripts to a separate folder and then just write to readme on where to copy them to folder root? That would help sort it out

**Comment by IlmariKu** [comment]
At least one issue regarding circular imports. I'll fix that

**Comment by machinewrapped** [comment]
Thanks, I'll take a look. 

I think I introduced some conflicts from the branch I just merged, sorry!

**Comment by IlmariKu** [comment]
I'll adding one more (for the common args-parsing) and I'll fix the conflicts!

What do you think in general about the changes? I personally just don't like repeating code or for example a cluttered root-level.

**Comment by machinewrapped** [comment]
I'm taking a look now. Consolidation is good - there is quite a bit of duplication and redundancy.

I'm not sure about moving the scripts into a subfolder. I think it's important that the setup process have as few steps as possible for new users. People aren't generally good at following instructions, so burying the scripts might cause confusion - I've given up on more than a few open source projects because the setup was opaque and error-prone, so I tried to minimize things that could go wrong.

I'm torn over having a single .cmd where needs the provider to be specified - it's cleaner, but it's an extra thing the user needs to know/remember to use the app successfully. If we keep it we should probably just name it 'subtrans.cmd', since GPT is just one of the supported options (the project name could have been chosen with more foresight!).

One possibility might be to have just the install scripts in the project root, and have them copy the matching .py/.cmd/.sh for each provider that the user installs to root?

It would probably be better to move the api_key argument out of the common script since not all providers will need an api key - e.g. Local Server, and it keeps provider-specific details together. Perhaps make description an argument to `CreateArgParser` as well.

File: PySubtitle/Helpers/prompts.py
Comment (machinewrapped) [inline PySubtitle/Helpers/prompts.py:1]: Wondering if it might be better to make these functions methods in `TranslationPrompt` so that people can override them if they subclass it. Some specialized models might require a specific format in future (https://github.com/machinewrapped/gpt-subtrans/issues/157)?

File: PySubtitle/Helpers/__init__.py [hunk 1]
```diff
@@ -488,10 +339,10 @@ def SanitiseSummary(summary : str, movie_name : str = None, max_summary_length :
 
     if len(summary) != original_len:
         logging.info(f"Summary was truncated from {original_len} to {len(summary)} characters")
-    
+
     return summary or None
 
-def FormatErrorMessages(errors : list[SubtitleError]):
+def FormatErrorMessages(errors : List[SubtitleError]):
```
Comment (machinewrapped) [inline PySubtitle/Helpers/__init__.py:345]: Oh, so that's how you do type hints for lists!

File: scripts/makedistro.bat [hunk 2]
```diff
@@ -6,4 +6,4 @@ pip install --upgrade -r requirements.txt
 pip install --upgrade openai
 pip install --upgrade google.generativeai
 pip install --upgrade anthropic
-pyinstaller --noconfirm --additional-hooks-dir="PySubtitleHooks" --add-data "theme/*;theme/" --add-data "instructions*;." --add-data "LICENSE;." --add-data "gui-subtrans.ico;." gui-subtrans.py
+pyinstaller --noconfirm --additional-hooks-dir="PySubtitleHooks" --add-data "theme/*;theme/" --add-data "instructions*;." --add-data "LICENSE;." --add-data "assets/gui-subtrans.ico;." gui-subtrans.py
```
Comment (machinewrapped) [inline scripts/makedistro.bat]: I think the add-data for instructions will need modifying to package them in a subdirectory... presumably something similar to theme (subtly different for windows/mac for no clear reason)

File: PySubtitle/Helpers/__init__.py (hunk 1 above)
Comment (rocketteering) [reply PySubtitle/Helpers/__init__.py:345]: Heh, yeah, it's totally not obvious with those types. I've just stumbled upon them via experience

File: scripts/makedistro.bat (hunk 2 above)
Comment (rocketteering) [reply scripts/makedistro.bat]: What do you think, should we move it to subdirectory?

File: PySubtitle/Helpers/prompts.py
Comment (rocketteering) [reply PySubtitle/Helpers/prompts.py:1]: I don't know as of now. They we're the first ones causing issues regarding circular-imports and I just felt the Helpers-folder was growing out of proportions

**Comment by IlmariKu** [comment]
> One possibility might be to have just the install scripts in the project root, and have them copy the matching .py/.cmd/.sh for each provider that the user installs to root?

I think this idea is great and you nailed it! This would keep the ease-of-use, but reduce the clutter from root.

Also, I do agree on your general sentiment about "People aren't generally good at following instructions, so burying the scripts might cause confusion". I've experienced the same too many times.

You're apparently running Windows, I'm presuming? I don't have access to a Windows system, but I could make the install-scripts for Mac (or just in general copy the required files).

File: scripts/makedistro.bat (hunk 2 above)
Comment (machinewrapped) [reply scripts/makedistro.bat]: Yes, I think it's a good idea to put them in a subdirectory :)

**Comment by machinewrapped** [comment]
> > One possibility might be to have just the install scripts in the project root, and have them copy the matching .py/.cmd/.sh for each provider that the user installs to root?
> 
> I think this idea is great and you nailed it! This would keep the ease-of-use, but reduce the clutter from root.
> 
> Also, I do agree on your general sentiment about "People aren't generally good at following instructions, so burying the scripts might cause confusion". I've experienced the same too many times.
> 
> You're apparently running Windows, I'm presuming? I don't have access to a Windows system, but I could make the install-scripts for Mac (or just in general copy the required files).

Yeah, I run Windows as my main system and have a Mac Mini for WFH that I use to prepare the macOS packages (but rarely run the app on it otherwise).

File: scripts/makedistro.bat (hunk 2 above)
Comment (rocketteering) [reply scripts/makedistro.bat]: Could you check this out that does the modifying need something to work? I mean, if you like in general would like to see this pr-merged at some point

**Comment by IlmariKu** [comment]
What do you think about this PR? Should I / we make the changes and try to get it in such a shape that we'd consider merging it? I'd agree on leaving the root for installs and could change that soon.

**Comment by machinewrapped** [comment]
> What do you think about this PR? Should I / we make the changes and try to get it in such a shape that we'd consider merging it? I'd agree on leaving the root for installs and could change that soon.

Yes, it looks good - we can do the install/root cleanup as a separate step and integrate the rest now :-)

**Comment by IlmariKu** [comment]
I'll remove them from the PR today and ping you afterwards!

File: scripts/makedistro.bat (hunk 2 above)
Comment (machinewrapped) [reply scripts/makedistro.bat]: I'll merge it now and fix up anything that needs fixing when I package the next release :-)

Comment (machinewrapped) [reply scripts/makedistro.bat]: Do you want to move the scripts back to root first, so we can tackle them separately later?

**Comment by IlmariKu** [comment]
Should be ready now @machinewrapped, I dropped the commits regarding moving the scripts

**Comment by machinewrapped** [comment]
I'll merge it tomorrow!

**Review by machinewrapped — APPROVED** [review]

**Comment by machinewrapped** [comment]
Alright, it's merged! 

I made a few changes to the way instructions files are loaded/saved first, so that it works in a PyInstaller package - and has cleaner dependencies (I hadn't noticed there was a dependency on GUI in the non-GUI module).

**Comment by IlmariKu** [comment]
Thanks @machinewrapped ! Looks super neat now, love that the root folder is tidied up now.
